src/request_data.zig

```diff
@@ -195,7 +196,54 @@ pub fn RequestData(comptime T: type) type {
                         null,
                     else => if (valid.optional(name)) |o| o.value else null,
                 },
-                .Pointer => (try valid.require(name)).value,
+                .Bool => {
+                    const item = try valid.require(name);
+                    if (item.value) |value| {
+                        if (eql(u8, value, "true")) {
+                            return true;
+                        }
+
+                        if (eql(u8, value, "false")) {
+                            return false;
+                        }
+
+                        return error.InvalidBool;
+                    }
+
+                    return error.UnexpectedNull;
+                },
+                .Int => {
+                    const item = try valid.require(name);
+                    if (item.value) |value| {
+                        return try std.fmt.parseInt(FieldType, value, 10);
```

Comment:
```suggestion
                        return try std.fmt.parseInt(FieldType, item.value orelse return error.UnexpectedNull, 10);
```

but perhaps valid.require shouldn't be able to return a null and an error?

Reply:
valid.require can't return a null but the item it returns could have a null value.